internal/listener/listener.go

```diff
@@ -46,7 +46,7 @@ func (l *Listener) ServeHTTP(rw http.ResponseWriter, req *http.Request) {
 	l.mux.ServeHTTP(rw, req)
 }
 
-func (l *Listener) Run() error {
+func (l *Listener) Run(ctx context.Context) error {
```

Comment:
Probably a good idea to document the new context + error return behavior of the function.